applysolutionswriter.cpp

```diff
@@ -36,37 +38,47 @@ ApplySolutionsWriter::~ApplySolutionsWriter()
 
 void ApplySolutionsWriter::WriteBandInfo(const std::string &name, const std::vector<Writer::ChannelInfo> &channels, double refFreq, double totalBandwidth, bool flagRow)
 {
-	_nChannels = channels.size();
-	_correctedData.resize(_nChannels*4);
-	
+	_nBandFineChannels = channels.size();
+	_correctedData.resize(_nBandFineChannels*4);
+
 	ForwardingWriter::WriteBandInfo(name, channels, refFreq, totalBandwidth, flagRow);
 
-	if(_nChannels < _nSolutionChannels || (_nChannels % _nSolutionChannels) != 0) {
+	// We permit situations where the total number of fine channels is > solution channels but only when
+	// the solution channels can be divided into the total number of fine channels evenly.
+	if (_nSolutionChannels < _nTotalFineChannels || (_nSolutionChannels % _nTotalFineChannels) != 0) {
 		std::ostringstream s;
-		s << "The provided solution file has an incorrect number of channels. Oobservation has " << _nChannels << " channels, and solution file has " << _nSolutionChannels << " channels.";
+		s << "The provided solution file has an incorrect number of channels. Observation has " << _nTotalFineChannels << " channels, and solution file has " << _nSolutionChannels << " fine channels.";
 		throw std::runtime_error(s.str());
 	}
 }
 
-void ApplySolutionsWriter::WriteRow(double time, double timeCentroid, size_t antenna1, size_t antenna2, double u, double v, double w, double interval, const std::complex<float>* data, const bool* flags, const float* weights)
+void ApplySolutionsWriter::WriteRow(double time, double timeCentroid, size_t antenna1, size_t antenna2, double u, double v, double w, double interval, const std::complex<float> *data, const bool *flags, const float *weights)
 {
-	// Apply solution to averaged data
-	int channelRatio = _nChannels / _nSolutionChannels;
-	
+	// Apply solution to averaged data or where resolution of cal solution is higher than the obs you are applying it to
```

Comment:
Can you clarify this comment? Do you mean something like `Apply solution to averaged data, taking into account that the resolution of cal solution might be higher than the obs you are applying it to` ?

Reply:
Hi Andre, yes, you got it- I'll update this comment to be more clear!